### DISCUSSION TITLE
Flickering when in edit mode

### DISCUSSION THREAD

**Discussion by luca-trifan** [forntoh/LcdMenu#177] [Q&A]
I have been working with LcdMenu on an ESP32 along with a standard 4x4 keypad and a 20x4 LCD display. Everything seems to be working fine, except when in edit mode, where the screen (especially the line that is being edited as well as the lines below it) start flickering. I can reduce the flickering partially by adding a delay in the loop, but that makes it sluggish. I have also tried using `processMenuCommand()` but the same thing happened. Furthermore, I tried using `isInEditMode()` to condition keypad polling with `keypad.waitForKey()`, but that just froze my program.

Is there anything I am missing?

```
void loop()
{
  //delay(200);
  char key = keypad.getKey();
  if (key != NO_KEY)
  {
    tone(BUZZER_PIN, 1700, 90);
    Serial.printf("Pressed key: %c\n", key);
  }
  if (key == 'D' && isMenuOn == 0)
  {
    menu.show();
    isMenuOn = 1;
  }
  if (key == 'A' && isMenuOn == 1)
  {
    menu.hide();
    isMenuOn = 0;
  }
  if (key == 'A')
    menu.back();
  else if (key == 'B')
    menu.up();
  else if (key == 'C')
    menu.down();
  else if (key == 'D')
    menu.enter();
  else if (key == '*')
    menu.left();
  else if (key == '#')
    menu.right();
  else
    menu.type(key);
  if (isMenuOn == 0)
      infoScreen();
}
```

`infoScreen()` is just a bunch of print statements I use for displaying some status information when the menu is not active

**Comment by forntoh** [comment]
@luca-trifan I haven't experienced this personally, I use the Uno for testing, nevertheless, there are is going to be a release in the next month or two (v4.3.0), in this version, drawing on the menu has be optimised so you should see an improvement. If after that you're still experiencing the same thing, please open a bug issue.

**Comment by forntoh** [comment]
If the issue is still present when using a latest release, please open an issue.
I'll close this conversation for now as I have been unable to reproduce this.
